### PR TITLE
(feat) added get_phi_vector_noise_obs() method 

### PR DESCRIPTION
**Pull request by mnfienen**
for ObservationEnsemble and a test. There is a test in there as well. Have a look.

### CODE REVIEW

**Comment by mnfienen** [comment]
Please don't merge yet - I'm working on some refactoring

File: pyemu/en.py [hunk 1]
```diff
@@ -811,15 +811,72 @@ def phi_vector(self):
...
+                    f"Could not open {noise_obs_filename}"
+                )
+        # make sure the index is all string (since the real)
+        noise_obs.index = [str(i) for i in noise_obs.index]
+        
+        # now make the calculations
+        cols = self._df.columns
+        weights = self.pst.observation_data.loc[cols, "weight"]
+        phi_vec = []
+        for idx in self._df.index.values:
```
Comment (briochh) [inline pyemu/en.py:872]: wonder if you could get away with:
`phi_vec = (((self._df-noise_obs) * weights)**2).dropna().sum(axis=1)`
pandas should align nicely on index and columns right?

Comment (mnfienen) [reply pyemu/en.py:872]: maybe, but @jtwhite79  pointed out that we need to handle inequality constraints, so gotta sum by groups and handle conditions about violation of inequality constraints. This led to some refactoring which may be close to done. If we get it working, I still like the idea of de-looping but not sure we can

Comment (briochh) [reply pyemu/en.py:904]: ah yea, good call. I think should be poss to work with a residual ensemble `res = (self._df-noise_obs)` and then zero out the residual that are the wrong sign for obs that are in inequality groups. `res.loc[:, ltobs] = res.loc[:, ltobs].clip(0)`

Comment (mnfienen) [reply pyemu/en.py]: that sounds like a good idea. I might try and look tomorrow. If you get a chance, maybe check out the latest commit I pushed. I think it's working, but too slow to be practical. It worked for a simple case without noise, but the noise obs example is a bigger file and def too slow. Late here but I can look tomorrow or Friday again

Comment (mnfienen) [reply pyemu/en.py:861]: All the tests are passing now so I think we are good, but the ensemble version is pretty slow. I like your idea @briochh so might try and implement that way

**Comment by briochh** [comment]
did you want to add more to this @mnfienen?

**Comment by mnfienen** [comment]
hey @briochh - I've been away for the month so missed this. I'm tempted to fold it in and experiment with the other option later unless you have time to test it?

**Comment by briochh** [comment]
Welcome back @mnfienen . Yeah go for the merge. We can look into testing some improves later from dev branch.

**Comment by mnfienen** [comment]
Right on @briochh  - merging

**Pull request merged**